pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -1032,8 +1033,12 @@ cdef class PyObjectHashTable(HashTable):
             val = values[i]
             hash(val)
 
-            if ignore_na and ((val != val or val is None)
-                              or (use_na_value and val == na_value)):
+            if ignore_na and (
```

Comment:
actualy you should use missing.pyx / checknull which has all of these conditions already and is our standard

Reply:
Ok, I'm assuming we still need the last condition `use_na_value and val == na_value` if somehow  a custom NA value is specified?

Reply:
It seems that `checknull` misses some things like `decimal.Decimal("nan")` that is caught by this logic:

```python
In [7]: val = decimal.Decimal("nan")                                                                                               

In [8]: val != val                                                                                                                 
Out[8]: True

In [9]: checknull(val)                                                                                                             
Out[9]: False
```

Reply:
decimal is not a first class citizen; you can create an issue about this, but tihs is a highly optimized check

Reply:
pls open a new issue an PR for this then for 1.1. this code duplication is horrendous.

Reply:
is there an issue for this?